src/main/scala/ai/diffy/proxy/HttpResponse.java

```diff
@@ -14,6 +14,9 @@ public HttpResponse(String status, HttpHeaders headers, String body) {
     }
 
     public String getStatus() {
+        if (status.contains(" ")) {
+            return status.split(" ")[0];
```

Comment:
```suggestion
            return status.trim().split("\\s+")[0];
```
I think this would be more robust and tolerant of formatting differences